pkg/controller/v1alpha2/applicationconfiguration/component.go

```diff
@@ -193,15 +193,17 @@ func (c *ComponentHandler) createControllerRevision(mt metav1.Object, obj runtim
 
 // get sorted controllerRevisions, prepare to delete controllerRevisions
 func sortedControllerRevision(appConfigs []v1alpha2.ApplicationConfiguration, revisions []appsv1.ControllerRevision,
-	revisionLimit int) (sortedRevisions []appsv1.ControllerRevision, toKill int, liveHashes map[string]bool) {
+	revisionLimit int, currentComponentName string) (sortedRevisions []appsv1.ControllerRevision, toKill int, liveHashes map[string]bool) {
 	liveHashes = make(map[string]bool)
 	sortedRevisions = revisions
 
 	// get all revisions used and skipped
 	for _, appConfig := range appConfigs {
 		for _, component := range appConfig.Spec.Components {
 			if component.RevisionName != "" {
-				liveHashes[component.RevisionName] = true
+				if ExtractComponentName(component.RevisionName) == currentComponentName {
```

Comment:
I don't think this is necessary, because we use label selector to filter the revisions, so the revisions passed into this function are all related to the specific Component only. https://github.com/crossplane/oam-kubernetes-runtime/blob/master/pkg/controller/v1alpha2/applicationconfiguration/component.go#L226


So there maybe no bug, right?

Reply:
yes, no bug. i close this issue.